packages/edit-site/src/components/template-actions/rename-menu-item.js

```diff
@@ -69,12 +73,7 @@ export default function RenameMenuItem( { template, onClose } ) {
 
 	return (
 		<>
-			<MenuItem
-				onClick={ () => {
-					setIsModalOpen( true );
-					setTitle( template.title.rendered );
-				} }
-			>
+			<MenuItem onClick={ () => setIsModalOpen( true ) }>
```

Comment:
I think there is no need to call `setState()` again when opening a popover, so I removed it.

Reply:
I believe this was set in #36879 so that unsaved changes wouldn't persist when closing and reopening the modal. It's a bit of an edge case, but if you make changes, click cancel, and then click "rename" again (without closing and reopening the parent menu of the "rename" button), you'll see the unsaved changes have been kept.